### ISSUE TITLE
Cannot consume scoped service from singleton

### ISSUE THREAD

**Issue by jlestercastro** [aspnet/AspNetCore#11171]
Hello, someone who can help me to fix this error.
I always got "System.InvalidOperationException: 'Cannot consume scoped service 'Lib.Data.LibContext' from singleton 'Lib.Data.RepositoryFactory'.'"

Where my Startup.cs is:

![image](https://user-images.githubusercontent.com/39945211/59404142-fc300000-8dd7-11e9-8f74-6bcd5ae2c0a9.png)

And my StartupHelper.cs is :

![image](https://user-images.githubusercontent.com/39945211/59404149-05b96800-8dd8-11e9-9748-522870988da0.png)

And i always got this error when i debug the project

![image](https://user-images.githubusercontent.com/39945211/59404159-0fdb6680-8dd8-11e9-8210-294752d41ade.png)

Someone can help me to fix this error.
Thank you in advance.

**Comment by davidfowl** [comment]
Yes this is blocked by default because it leads to components leaks. While it's not 100% wrong in certain cases, we block this by default because it's usually a result of buggy code. In your case ,it seems like you need to create a scope before resolving the scoped service.
